fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/AccessWidenerGenerator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.datagen;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import net.minecraft.data.server.RecipesProvider;
+
+public class AccessWidenerGenerator {
```

Comment:
Why is the AW generator for the *main* sourceset in the *testmod*?

Reply:
This is going to move all into gradle, the reason for putting it here is it doesnt need to get shipped with the mod (as its only used by the buildscript)